tokio/src/runtime/handle.rs

```diff
@@ -61,6 +61,15 @@ impl Handle {
     pub fn current() -> Self {
         context::current().expect("not currently running on the Tokio runtime.")
     }
+
+    /// Returns a Handle view over the currently running Runtime
+    ///
+    /// Returns `None` if no Runtime has been started
+    ///
+    /// Contrary to `current`, this never panics
+    pub fn try_current() -> Option<Self> {
```

Comment:
I definitely have a slight preference for @carllerche's suggestion that this return a `Result`...the error type could just format with the message "not currently running on the Tokio runtime.", like the `expect` in `current()`. Not a big deal, though, IMO.